resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -130,12 +197,27 @@ private[spark] class KubernetesClusterSchedulerBackend(
     super.start()
     executorWatchResource.set(kubernetesClient.pods().withLabel(SPARK_APP_ID_LABEL, applicationId())
       .watch(new ExecutorPodsWatcher()))
+
+    podAllocationInterval.foreach(allocator.scheduleWithFixedDelay(allocatorRunnable,
+      0,
```

Comment:
I think since now `podAllocationInterval` is always going to be provided (we always set it to `Some(...)` or throw an exception) then this thread will always be running. Is this the intended behavior? If so, no need to use foreach and options here.